src/main/java/core/basesyntax/strategy/operation/PurchaseOperation.java

```diff
@@ -0,0 +1,13 @@
+package core.basesyntax.strategy.operation;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void performOperation(FruitTransaction transaction) {
+        int currentQuantity = Storage.storage.get(transaction.getFruitName());
+        Storage.storage.put(transaction.getFruitName(),
+                currentQuantity - transaction.getQuantity());

```

Comment:
Handling Purchase operation. Check result balance before saving it in the Storage - it should be positive. Throw `RuntimeException` in case the balance is negative. Right now, there is no check for a negative balance, which could lead to incorrect storage state.